PageObjectPattern/src/test/java/apiSerenityStep/SpotifyApiStep.java

```diff
@@ -1,114 +1,36 @@
-    }
-
-//to check one field in response
-    @Step("check valid data in Get response")
-    public void checkIfDataIsValidGet(String requiredField, String responseData) {
-
-        responseGet.then().assertThat().body(requiredField, equalTo(responseData));
-
-    }
-
-    @Step("sending POST request to create Playlist")
-    public void createPlayList(String endpoint, String playListName) {
-
-        responsePost = SerenityRest.given()
-                .body("{\n" + " \"name\":\"" + playListName + "\",\n" + " \"public\":\"true\"\n" + "}")
-                .header("Authorization", "Bearer " + accessToken).header("Accept", "string").post(baseUrl + endpoint);
-
-    }
-
-//to check one field in response
-    @Step("check Post response and return boolean result")
-    public boolean checkResponseAfterPost(String playListName) {
-
-        String fieldName = responsePost.getBody().jsonPath().getString("name");
-        boolean result;
-        result = (fieldName.equals(playListName));
-        return result;
-    }
-
-//to check many fields in response
-    @Step("check multiple fields in Post response")
-    public void checkFieldsInPostResponse(String playListName) {
-
-        responsePost.then().assertThat().body("name", equalTo(playListName)).body("owner.id", equalTo("czc11mg48dvwd16aeq0jofr3j")).body("owner.display_name",
-                equalTo("Yulia")).body("tracks.limit", equalTo(100));
-
-    }
-
-    @Step("check presense of required fields")
-    public boolean checkPresenseOfRequiredFields(String requiredField) {
-
-        ResponseBody body = responseGet.getBody();
-        String bodyStringValue = body.asString();
-        boolean result;
-        result = (bodyStringValue.contains(requiredField));
-        return result;
-
-    }
-
-    @Step("invalidate access token")
-    public void invalidateAccessToken() {
-        this.accessToken = "123";
+import java.util.HashMap;
+import java.util.Map;
+
+// This class is used to customize api calls
+public class SpotifyApiStep extends BaseStep {
+
+    // getting one field
+    @Step("getting one field from response")
+    public String checkPresenseOfRequiredFields() {
+        String oneString = responseGet.getBody().jsonPath().getString("albums.limit");
+        return oneString;
+    }
+
+    @Step("sending POST request to create Playlist, endpoint: /v1/users/{userid}/playlists")
+    public void postPlayList(String endpoint, String playListName) {
+        String accessToken = Serenity.sessionVariableCalled("token").toString();
+        System.out.println("sending POST request with this token" + accessToken);
+        Map<String, Object> map = new HashMap<>();
+        map.put("name", playListName);
+        map.put("public", "true");
+        responsePost = SerenityRest.given().body(map).header("Authorization", "Bearer " + accessToken)
+                .header("Accept", "string").post(baseUrl + endpoint);
+    }
+
+    @Step("sending get request with app token and MAX Limit")
```

Comment:
Description might be more specific since it's not a general GET method, but a request to a specific endpoint.